### PR TITLE
Cleanup: simplify tests in validating finalizer creating

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Forget to address the ToDo.

### CODE REVIEW

File: test/integration/controller/core/clusterqueue_controller_test.go [hunk 1]
```diff
@@ -210,11 +210,9 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 		})
 
 		ginkgo.It("Should add a finalizer to the clusterQueue", func() {
-			gomega.Eventually(func() []string {
-				var newCQ kueue.ClusterQueue
-				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
-				return newCQ.GetFinalizers()
-			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{kueue.ResourceInUseFinalizerName}))
+			var newCQ kueue.ClusterQueue
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply test/integration/controller/core/clusterqueue_controller_test.go]: make sense

File: test/integration/controller/core/resourceflavor_controller_test.go [hunk 2]
```diff
@@ -81,14 +71,9 @@ var _ = ginkgo.Describe("ResourceFlavor controller", func() {
...
-			gomega.Eventually(func() []string {
-				var newRF kueue.ResourceFlavor
-				if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &newRF); err != nil {
-					return nil
-				}
-				return newRF.GetFinalizers()
-			}, framework.Timeout, framework.Interval).Should(utiltesting.Equal([]string{kueue.ResourceInUseFinalizerName}))
+			var newRF kueue.ResourceFlavor
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(&rf), &newRF)).To(gomega.Succeed())
+			gomega.Expect(rf.GetFinalizers()).Should(utiltesting.Equal([]string{kueue.ResourceInUseFinalizerName}))
```
Comment (alculquicondor) [inline test/integration/controller/core/resourceflavor_controller_test.go:76]: I would just remove this entire `It`.

Comment (kerthcet) [reply test/integration/controller/core/resourceflavor_controller_test.go]: done

**Pull request merged**